synapse/spam_checker_api/__init__.py

```diff
@@ -26,18 +27,18 @@ class SpamCheckerApi(object):
     access to rooms and other relevant information.
     """
 
-    def __init__(self, hs):
+    def __init__(self, hs: "synapse.server.HomeServer"):
         self.hs = hs
 
         self._store = hs.get_datastore()
 
     @defer.inlineCallbacks
-    def get_state_events_in_room(self, room_id, types):
+    def get_state_events_in_room(self, room_id: str, types: tuple) -> defer.Deferred:
```

Comment:
generally we've been declaring bankruptcy when trying to declare the return type of `@defer.inlineCalbacks` functions; the problem being that it's only the decorated version of the function that actually returns a Deferred and mypy etc get confused when you try to `return state.values()` instead. 

Instead we either convert to `async/await` first, or just leave the return type unannotated.

Reply:
Interesting. I can just remove the return type for now if that matches what is done in other places.

Reply:
it doesn't matter. I'd just land it.